From 4e6a75eed3602f10aff516f49c8088c6da8db9df Mon Sep 17 00:00:00 2001
From: Samuel Martin <s.martin49@gmail.com>
Date: Sun, 31 Aug 2014 12:11:57 +0200
Subject: [PATCH 1/1] cmake: use the standard CMake flag to drive the shared
 object build

If BUILD_SHARED_LIBS is set and ENABLE_STATIC undefined, then drive
ENABLE_STATIC with the BUILD_SHARED_LIBS value.

Signed-off-by: Samuel Martin <s.martin49@gmail.com>
---
 CMakeLists.txt | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 317ffa1..9931957 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,17 @@ project(taglib)
 
 cmake_minimum_required(VERSION 2.6.0 FATAL_ERROR)
 
-option(ENABLE_STATIC "Make static version of libtag"  OFF)
+# Use the standard CMake flag to drive the shared object build.
+if(DEFINED BUILD_SHARED_LIBS AND NOT DEFINED ENABLE_STATIC)
+  if(BUILD_SHARED_LIBS)
+    set(ENABLE_STATIC OFF)
+  else()
+    set(ENABLE_STATIC ON)
+  endif()
+else()
+  option(ENABLE_STATIC "Make static version of libtag"  OFF)
+endif()
+
 if(ENABLE_STATIC)
   add_definitions(-DTAGLIB_STATIC)
   set(BUILD_SHARED_LIBS OFF)
-- 
2.1.0

